ResetOptionsBinder.cs

```diff
@@ -0,0 +1,55 @@
+﻿using AMSMigrate.Contracts;
+using FFMpegCore.Enums;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Binding;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AMSMigrate
+{
+    internal class ResetOptionsBinder : BinderBase<ResetOptions>
+    {
+        private readonly Option<string> _sourceAccount = new Option<string>(
+           aliases: new[] { "--source-account-name", "-n" },
+           description: "Azure Media Services Account.")
+        {
+            IsRequired = true,
+            Arity = ArgumentArity.ExactlyOne
+        };
+
+        private readonly Option<bool> _all = new Option<bool>(
+           aliases: new[] { "--all", "-a" },
+           description: "Reset all assets in the account, regardless of their migration status.")
+        {
+            IsRequired = false
+        };
+        private readonly Option<bool> _failed = new Option<bool>(
+          aliases: new[] { "--failed", "-f" },
+          description: "Reset the failed migrated assets in the account, This is the default setting.")
+        {
+            IsRequired = false
+        };
```

Comment:
Instead of two options in the command line, I would suggest to have a single option like --scope [value],  the value could be All or Failed.  the default value is All.

All means all assets that meets the filter condition (if filter is added).

Failed means only the failed assets that in the input selection with filter-applied if filter is there.

With two separate options like this one, you will have to figure out a right behavior if both -a and -f are set in the command line.